scripts/create_tarred_transformer_lm_dataset.py

```diff
@@ -0,0 +1,318 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Execution :
+
+python create_tarred_tokenized_text_lm_dataset.py \
+    --text_path=<comma seperated text filepaths> \
+    --data_root=<path to output directory> \
+    --tokenizer_name="bert-base-cased" \
+    --tokenizer_vocab_file=<path to vocab file for tokenizer> \
+    --num_shards=64 \
+    --chunk_size=8192 \
+    --chunk_write_buffer=512 \
+    --lower_case \
+    --log
+"""
+
+
+import argparse
+import glob
+import json
+import logging
+import os
+import tarfile
+
+import joblib
+import numpy as np
+from tqdm import tqdm
+
+from nemo.collections.nlp.modules.common.tokenizer_utils import get_tokenizer
+
+parser = argparse.ArgumentParser(description='Tarred Tokenized dataset for text language modelling')
+
+# Data path arguments
+parser.add_argument('--text_path', required=True, default=None, type=str, help='Text paths, seperated by commas')
+parser.add_argument('--data_root', required=True, default=None, type=str, help='Output directory')
+
+# General arguments
+parser.add_argument(
+    '--chunk_write_buffer',
+    default=128,
+    type=int,
+    help='Number of chunks of `chunk_size` to buffer for parallel tokenization and serial write to disk',
+)
+parser.add_argument('--lower_case', action='store_true', help='Whether to lower case the corpus')
```

Comment:
Why do we need this as a separate option? I think that this is a property of specific tokenizer.